caiman/cluster.py

```diff
@@ -406,7 +407,12 @@ def setup_cluster(backend='multiprocessing', n_processes=None, single_thread=Fal
                 raise Exception(
                     'A cluster is already runnning. Terminate with dview.terminate() if you want to restart.')
             c = None
-            dview = Pool(n_processes)
+            if platform.system() == 'Darwin':
+                mp_start_method = 'forkserver'  # default 'fork' crashes multi-threaded BLAS on Mac
+            else:
+                mp_start_method = None # use default for platform
+            ctx = multiprocessing.get_context(mp_start_method)
```

Comment:
Hi, with some configurations, this get_context is illegal (which is where it's breaking the Linux part of our Jenkins setup). If you could hoist this version of it into the Darwin block and have the old call to Pool (without  the ctx) in the else block, that'd prevent breakage on those platforms.

Reply:
Actually, this is breaking CI on OSX too. For our Python2/OSX, it fails with the get_context like our Linux tests do.

For Python3/OSX, it's failing with a data error:

```  File "/Users/jenkins/workspace/CaImAn_PR-208-DB3OXXZWJBG3JQLECJIW2BRE7TXMWHMCJYICGXB2PBZQFDBHNTQQ/caiman/source_extraction/cnmf/cnmf.py", line 508, in fit
    Yr, A, b, C, f, dview=self.dview, bl=None, c1=None, sn=None, g=None, **options['temporal_params'])
  File "/Users/jenkins/workspace/CaImAn_PR-208-DB3OXXZWJBG3JQLECJIW2BRE7TXMWHMCJYICGXB2PBZQFDBHNTQQ/caiman/source_extraction/cnmf/temporal.py", line 232, in update_temporal_components
    ITER, YrA, c1, sn, g, Cin, T, nA, dview, debug, AA, kwargs)
  File "/Users/jenkins/workspace/CaImAn_PR-208-DB3OXXZWJBG3JQLECJIW2BRE7TXMWHMCJYICGXB2PBZQFDBHNTQQ/caiman/source_extraction/cnmf/temporal.py", line 371, in update_iteration
    constrained_foopsi_parallel, args_in).get(4294967)
  File "/Users/jenkins/workspace/CaImAn_PR-208-DB3OXXZWJBG3JQLECJIW2BRE7TXMWHMCJYICGXB2PBZQFDBHNTQQ/test/osx-python3/lib/python3.5/multiprocessing/pool.py", line 644, in get
    raise self._value
numpy.linalg.linalg.LinAlgError: Array must not contain infs or NaNs```

(I'm assuming you can't see our Jenkins)